File: high_prime_factor.rb
#!/usr/local/bin/ruby

def isPrime(m_num)
  m_sq = Math.sqrt(m_num).to_i
  (2..m_sq).each do |m_fact|
    return false if ( m_num % m_fact == 0 )
  end
  return true
end

def highPrimeFactor(m_num)
  m_sq = Math.sqrt(m_num).to_i
  (m_sq.downto(1)).each do |m_fact|
    return m_fact if ( m_num % m_fact == 0 && isPrime(m_fact) )
  end
end

puts highPrimeFactor(26168960)

File: high_prime_factor.py
#!/usr/bin/python

import math

def isPrime(m_num):
  m_sq = int(math.sqrt(m_num))
  for m_fact in xrange(2, m_sq):
    if ( m_num % m_fact == 0 ): return False
  return True

def highPrimeFactor(m_num):
  m_sq = int(math.sqrt(m_num))
  print m_sq
  for m_fact in xrange(m_sq, 1, -1):
    if ( m_num % m_fact == 0 and isPrime(m_fact) ): return m_fact

print highPrimeFactor(600851475143)
